pylabrobot/liquid_handling/liquid_handler.py

```diff
@@ -1760,15 +1761,18 @@ async def dispense96(
       if not len(containers) == 96:
         raise ValueError(f"dispense96 expects 96 wells, got {len(containers)}")
 
-      for channel, well in zip(self.head96.values(), containers):
-        # even if the volume tracker is disabled, a liquid (None, volume) is added to the list
-        # during the aspiration command
-        liquids = channel.get_tip().tracker.remove_liquid(volume=volume)
-        reversed_liquids = list(reversed(liquids))
-        all_liquids.append(reversed_liquids)
+      for well, channel in zip(containers, self.head96.values()):
+        # check if volume tracking is disabled
+        if well.tracker.is_disabled or not does_volume_tracking():
+          reversed_liquids = [(None, volume)]
+          all_liquids.append(reversed_liquids)
+        else:
+          liquids = channel.get_tip().tracker.remove_liquid(volume=volume)
+          reversed_liquids = list(reversed(liquids))
+          all_liquids.append(reversed_liquids)
 
-        for liquid, vol in reversed_liquids:
-          well.tracker.add_liquid(liquid=liquid, volume=vol)
+          for liquid, vol in reversed_liquids:
+            well.tracker.add_liquid(liquid=liquid, volume=vol)
```

Comment:
this was likely part of the actual issue: the liquids weren't added to each well (just the last one).

Reply:
it should have that condition! for putting liquid into wells. that part is conditional on volume tracking being enabled

but or channels we always need it :)

> this is what i thought caused the issue since turning volume tracking off would prevent aspirate96 from tracking liquids

in this case, asp96 would actually still track liquids.

the problem in this line (and i think the root cause of the issue you describe in the PR message) is that liquid doesn't get added to well on dispense. the code you change in these 2 highlighted lines was obviously wrong before